src/perftest_parameters.c

```diff
@@ -961,9 +961,15 @@ static void force_dependecies(struct perftest_parameters *user_param)
 	}
 
 	if (user_param->post_list > 1) {
-		user_param->cq_mod = user_param->post_list;
-		printf(RESULT_LINE);
-		printf("Post List requested - CQ moderation will be the size of the post list\n");
+		if (user_param->cq_mod == DEF_CQ_MOD) {
+			user_param->cq_mod = user_param->post_list;
+			printf(RESULT_LINE);
+			printf("Post List requested - CQ moderation will be the size of the post list\n");
+		} else if ((user_param->post_list % user_param->cq_mod) != 0) {
```

Comment:
Why must post list size be a multiple of CQ moderation? 
Note that in default case CQ mod = 100, post_list = 1, but 1 % 100 != 0 and it is just fine.

CQ moderation is how often we request a CQE per WQE, post_list is how often we ring the doorbell per WQE.

Reply:
Previous design was to request a CQE on last WQE of post list.
My original proposal is to be able to request multiple CQEs per post list, but for that to work, we need a CQE on the last WQE of the post list. Otherwise, the test might get stuck if last posted list doesn't end with a CQE.

But on second thought after your comment, I think the condition should be that #iterations is a multiple of CQ mod to guarantee no hangs. Do you prefer to have this check explicitly or same as post list?

From README/Known Issues:
>  8. for postlist, tests will not finish correctly when using iterations and num_iters/poslist_val has a remainder.

Reply:
Agreed with #iterations is a multiple of CQ mod and a multiple of postlist_val. Explicit checks and error messages are better from my viewpoint.

> the test might get stuck if last posted list doesn't end with a CQE.
it was not expected by me, are you sure? Cause if postlist_val == 1 we are good with any CQE mod. What should be the difference with postlist_val == 2 and CQE mod == 100?

Reply:
Are we good with any CQE mod even if it doesn't align with #iterations? e.g. CQE mod == 100 and #iterations == 1024. The last CQE that we get is on the 1000th WR. What happens with the last 24?

Reply:
> What happens with the last 24?

I need to investigate it, to be 100% sure. But I guess in current perftest we just don't get a CQ in such case and get out of the loop by another condition.

Reply:
I thought about generalizing the CQ moderation but I think it will have a performance penalty.
Let's take for example post_list of size 4, and CQE every 3 WQEs. In such case, I can't do anything during init time (`ctx_set_send_reg_wqes()`) and in run time I will have to keep turning on and off the SIGNAL bit over all the post_list.

Also regarding my question:

> What happens with the last 24?

Current code always requests CQE on last iteration.

What do you suggest?

Reply:
1) It is not bad that perftest can mimic an unoptimized case. We just need to make sure that it is not the default one :)
2) regarding the last question: looks like for this reason we have a condition on line 3823 to generate a CQe on the last iteration.

Reply:
Adding code to handle the generic case affects the default one: whenever we will have `post_list > 1`, we will have to check the signal bit on all WRs.

Reply:
Not sure it can hurt the performance dramatically, since it is done only if the condition is hit.
Anyway, thank you for a great work, I will test it soon and merge if it works well!